testing/regress/ecl/persist_refresh.ecl

```diff
@@ -0,0 +1,67 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2015 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+import Std;
+import Std.File AS FileServices;
+
+//
+//version persistRefresh=true
+//version persistRefresh=false
+
+import ^ as root;
+
+string OriginalTextFilesIp := '.' : STORED('OriginalTextFilesIp');
+string OriginalTextFilesEclPath := '' : STORED('OriginalTextFilesEclPath');
+string OriginalTextFilesOsPath := '' : STORED('OriginalTextFilesOsPath');
+
+persistRefresh := #IFDEFINED(root.persistRefresh, true);
+
+#IF (persistRefresh=true)
+InputFile := OriginalTextFilesOsPath + '/download/europe1.txt';
+#ELSE
+InputFile := OriginalTextFilesOsPath + '/download/europe2.txt';
+#END
+
+ClusterName := 'mythor';
+DestFile := '~regress::europe.txt';
+ESPportIP := 'http://127.0.0.1:8010/FileSpray';
+
+spray := FileServices.SprayVariable(OriginalTextFilesIp,
+                                     InputFile,
+                                     destinationGroup := ClusterName,
+                                     destinationLogicalName := DestFile,
+                                     espServerIpPort := ESPportIP,
+                                     ALLOWOVERWRITE := true);
+
+rec := RECORD
+    string country;
+    integer4 population;
+END;
+
+ds := DATASET(DestFile, rec, CSV(HEADING(1), SEPARATOR(','), QUOTE([])));
+
+CountriesDS := ds:PERSIST('~REGRESS::PersistRefresh', SINGLE, REFRESH(persistRefresh));
+
+OutputResults := OUTPUT(CountriesDS);
+
+pause := STD.System.Debug.Sleep(2000);
+
+sequential (
+       spray;
```

Comment:
Ok, so here's my concern worded and then re-worded - sorry concepts aren't serialized in my head, takes a few iterations sometimes ;)

The actual c++ code changes etc show me that the check to rebuild the persist is in eclagent but when does this occur within the ecl? the langref snippet I found "By definition, PERSIST on an attribute means the attribute is evaluated outside of any given evaluation order. Therefore, SEQUENTIAL has no effect on PERSISTed attributes." has made me wonder about it's declarative nature and therefore I was wondering if, in the given test it was making the comparison to  that which intended? I've never really sprayed anything before but the ALLOWOVERWRITE option implies to me that a sprayed file persists on disk and since the logical filename is static, if control flow is up in the air, then could it be possible that the persist was making a comparison to the old sprayed file (not the new one that is diff) since the new sprayed file is within the sequential and I'm not sure what happens with the evaluation of a persist (as the lang ref snippet as made me wonder)? Sorry, this may not follow too well.

perhaps paraphrasing the langref snippet makes me wonder if OutputResults is hoisted out of the SEQUENTIAL and possibly evaluated along with all its dependencies which is just
ds := DATASET(DestFile, rec, CSV(HEADING(1), SEPARATOR(','), QUOTE([])));
rec & DestFile.
"spray" isn't a dependency and is in the sequential, if OutputResults is evaluated before the sequential, before the spray, then it would be reading the old file that is identical to the persisted file and therefore refresh is not being tested at all.